apps/internal/base/base.go

```diff
@@ -351,12 +352,20 @@ func (b Client) AcquireTokenSilent(ctx context.Context, silent AcquireTokenSilen
 	if silent.Claims == "" {
 		ar, err = AuthResultFromStorage(storageTokenResponse)
 		if err == nil {
-			if b.shouldRefresh(storageTokenResponse.AccessToken.RefreshOn.T, tenant) {
+			if b.shouldRefresh(storageTokenResponse.AccessToken.RefreshOn.T) {
 				b.refreshMu.Lock()
-				b.removeTenantFromCanRefresh(tenant)
-				defer b.refreshMu.Unlock()
-				if tr, er := b.Token.Credential(ctx, authParams, silent.Credential); er == nil {
-					return b.AuthResultFromToken(ctx, authParams, tr)
+				if _, exists := b.canRefresh[tenant]; !exists {
+					var empty atomic.Value
+					empty.Store(false)
+					b.canRefresh[tenant] = &empty
+				}
+				refreshValue := b.canRefresh[tenant]
```

Comment:
you can do this with one read
```suggestion
				refreshValue, ok := b.canRefresh[tenant]
				if !ok {
					refreshValue = &atomic.Value{}
					refreshValue.Store(false)
					b.canRefresh[tenant] = refreshValue
				}
```
(also, `ok` is the idiomatic name for the second variable in this pattern, which is called "comma ok")

Reply:
Noted, will use `ok`